### PR TITLE
Upgrade to Netbeans 12 

### PR DESCRIPTION
**Pull request by aymanhab**
Fixes issue #1290 

### Brief summary of changes
Use Netbeans 12.3 instead of discontinued 8.2 for the platform the GUI is based upon and for actual building on ci

### CHANGELOG.md (choose one)

- no need to update because...
- updated...

### CODE REVIEW

**Comment by aymanhab** [comment]
@carmichaelong @nickbianco @jenhicks If you can give the artifact of this PR a spin would be great. Things to pay attention to:
1. Installation (Icons)
2. Toolbar and text size/layout
3. Resolution of the graphics window (latest netbeans is DPI aware with pros and cons)

**Comment by carmichaelong** [comment]
@aymanhab Looking good on Windows so far. Installation was good. Font seems about right for my setup. I also tried some standard things that worked as expected:
- loaded gait2392 with normal.mot motion
- loaded gait10dof18 with CMC results
- performed OpenSense pipeline
- tested plotting forces from files given with the distribution

**Comment by aymanhab** [comment]
@mitkof6 @adamkewley or @halleysfifthinc Not sure if you'd be interested in this as I don't have access to an Ubuntu machine to fix the github action script.

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -275,3 +273,113 @@ jobs:
...
+
+    - name: Update submodules
+      run: git submodule update --init --recursive -- opensim-models opensim-visualizer Gui/opensim/threejs
+
+    - name: Build GUI
+      id: build-gui
+      run: |
+        mkdir build
+        cd build
+        cmake ../ -DCMAKE_PREFIX_PATH=~/opensim-core-install -DANT_ARGS="-Dnbplatform.default.netbeans.dest.dir=$HOME/netbeans-12.3;-Dnbplatform.default.harness.dir=$HOME/netbeans-12.3/harness"
```
Comment (halleysfifthinc) [inline .github/workflows/continuous-integration.yml:369]:
```suggestion
        cmake ../ -DCMAKE_PREFIX_PATH=~/opensim-core-install -DANT_ARGS="-Dnbplatform.default.netbeans.dest.dir=$HOME/netbeans-12.3/netbeans;-Dnbplatform.default.harness.dir=$HOME/netbeans-12.3/netbeans/harness"
```

**Review by halleysfifthinc — CHANGES_REQUESTED** [review]
Looks like you missed the nesting of the netbeans install folder for Linux. Even with that, I am unable to successfully build, I get errors about `org.openide.awt.ActionRegistration missing element enabledOn`. I have zero-familiarity with Java, so I don't know where to go from there. The fact that it builds successfully on Mac/Windows suggests to me that it might be a Java version/distribution issue? I'm running:

```sh
$ java -version
openjdk version "1.8.0_282"
OpenJDK Runtime Environment (build 1.8.0_282-8u282-b08-0ubuntu1~20.04-b08)
OpenJDK 64-Bit Server VM (build 25.282-b08, mixed mode)
```

**Comment by aymanhab** [comment]
Thanks much @halleysfifthinc I applied your change and will see how the log compares to osx build.

**Comment by aymanhab** [comment]
Build is now successful but script to build tar fails because case sensitive name opensim.zip doesn't match OpenSim.zip

File: Gui/opensim/build.xml [hunk 2]
```diff
@@ -331,9 +331,9 @@
                 includes="**/jxbrowser-*mac*.jar"/>
             <fileset dir="${opensim.dist.dir}/OpenSim/modules/ext"
                 includes="**/jxbrowser-*win*.jar"/>
-            <fileset file="${opensim.dist.dir}/bin/OpenSim.exe"/>
+            <fileset file="${opensim.dist.dir}/bin/opensim.exe"/>
```
Comment (halleysfifthinc) [inline Gui/opensim/build.xml:334]: Not sure that the .exe name case changed.

**Review by halleysfifthinc — CHANGES_REQUESTED** [review]
On line 322 of build.xml, we can inline the "unzip-distro" step as follows:

```xml
<target name="build-linux" depends="copy-visualizer,copy-models">
        <!-- <echoproperties/> -->
        <unzip src="dist/opensim.zip" dest="dist/installer" />
```

File: Gui/opensim/build.xml (hunk 2 above)
Comment (aymanhab) [reply Gui/opensim/build.xml:334]: @halleysfifthinc The name actually changed on windows to opensim.exe so I don't think it's failing to delete the file. I'll apply your inlining suggestion. Thank you 👍

**Comment by halleysfifthinc** [comment]
My last suggestion missed that `unzip-distro` depended on `build-zip`, and now it doesn't actually (attempt to) build the project. :upside_down_face: 

After I remedied that there were a couple other `OpenSim` => `opensim` case changes to fix:

```diff
diff --git a/Gui/opensim/build.xml b/Gui/opensim/build.xml
index 466ee962..4e4df3fb 100644
--- a/Gui/opensim/build.xml
+++ b/Gui/opensim/build.xml
@@ -32,7 +32,7 @@
     <condition property="opensim.dist.dir" value="dist/OpenSim.app/Contents/Resources/OpenSim">
         <os family="mac"/>
     </condition>
-    <condition property="opensim.dist.dir" value="dist/installer/OpenSim">
+    <condition property="opensim.dist.dir" value="dist/installer/opensim">
         <and>
             <os family="unix"/>
             <not>
@@ -319,7 +319,7 @@
         </exec>

     </target>
-    <target name="build-linux" depends="copy-visualizer,copy-models">
+    <target name="build-linux" depends="build-zip,copy-visualizer,copy-models">
         <!-- <echoproperties/> -->
         <unzip src="dist/opensim.zip" dest="dist/installer" />
         <copy todir="${opensim.dist.dir}/Resources/Code/GUI">
@@ -327,12 +327,12 @@
         </copy>

         <delete>
-            <fileset dir="${opensim.dist.dir}/OpenSim/modules/ext"
+            <fileset dir="${opensim.dist.dir}/opensim/modules/ext"
                 includes="**/jxbrowser-*mac*.jar"/>
-            <fileset dir="${opensim.dist.dir}/OpenSim/modules/ext"
+            <fileset dir="${opensim.dist.dir}/opensim/modules/ext"
                 includes="**/jxbrowser-*win*.jar"/>
             <fileset file="${opensim.dist.dir}/bin/opensim.exe"/>
-            <fileset file="${opensim.dist.dir}/bin/OpenSim64.exe"/>
+            <fileset file="${opensim.dist.dir}/bin/opensim64.exe"/>
             <fileset file="dist/opensim.zip"/>
         </delete>

@@ -354,7 +354,7 @@
         <zip basedir="${opensim.dist.dir}/Resources/" destfile="${opensim.dist.dir}/Resources.zip" />
         <delete dir="${opensim.dist.dir}/Resources"/>

-        <replace file="${opensim.dist.dir}/etc/OpenSim.conf" value="${app.version}">
+        <replace file="${opensim.dist.dir}/etc/opensim.conf" value="${app.version}">
             <replacefilter token="@VERSION@" />
         </replace>
```

I am now able to successfully build. Font size and spacing matches the system font better now!

**Comment by aymanhab** [comment]
Awesome, thanks @halleysfifthinc Please commit your changes as above and this PR will be good to go.

**Comment by carmichaelong** [comment]
@aymanhab I merged @halleysfifthinc 's PR into this branch

**Comment by aymanhab** [comment]
Thanks @carmichaelong so are we good to merge?

**Review by carmichaelong — APPROVED** [review]

**Comment by aymanhab** [comment]
Great, thank you @carmichaelong and @halleysfifthinc Merging

**Pull request merged**